File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/NetworkServices.kt
package vn.hitechcloud.sdk

/** Network Services: IP addresses and reverse DNS for services */
class NetworkServicesResource(private val httpClient: HttpClient) {
    /** GET /api/service/{id}/ip - List Service IP Addresses */
    fun listIps(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/ip", params)
    }

    /** GET /api/service/{id}/rdns - Get reverse DNS entries */
    fun getRdns(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/rdns", params)
    }

    /** POST /api/service/{id}/rdns - Update reverse DNS entries */
    fun updateRdns(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/service/{id}/rdns", params)
    }

}

File: sdk/swift/Sources/HiTechCloudSDK/NetworkServices.swift
import Foundation

/// Network Services: IP addresses and reverse DNS for services
public class NetworkServicesResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/service/{id}/ip - List Service IP Addresses
    public func listIps(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/ip", params: params)
    }

    /// GET /api/service/{id}/rdns - Get reverse DNS entries
    public func getRdns(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/rdns", params: params)
    }

    /// POST /api/service/{id}/rdns - Update reverse DNS entries
    public func updateRdns(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/service/{id}/rdns", params: params)
    }

}
